### ISSUE TITLE
Use vst instead of lxvst

### ISSUE THREAD

**Issue by falkTX** [flathub/studio.kx.carla#2]
Not sure why `lxvst` needs to be used or where it even comes from.
Carla has always used vst to refer to native things, feels very weird and awkward to treat it different on Linux. Should BSD be the same? Haiku, GNU etc? Plugins are loaded natively, and VST works fine on all platforms (not that official SDK supports that in written form, but we do not care or use that anyway).

We should get rid of the `lx` prefix that does nothing but bring confusion.

**Comment by Bleuzen** [comment]
VST plugins are built into `/app/extensions/Plugins/lxvst` on flathub. I just followed the documentation for the plugin base extension:
https://github.com/flathub/org.freedesktop.LinuxAudio.BaseExtension/tree/branch/21.08

As for why it is called exactly this way, maybe @hfiguiere could comment.

**Comment by falkTX** [comment]
Ok I am interested to know more. From what I am aware only Ardour ever used that name and it was only because it supported Windows VSTs before Linux ones, sometimes on the same build, so it went with lxvst to not get Windows and Linux VSTs mixed up on the same folder(s).

Flatpak never supported Windows VSTs, correct? Then the use of lxvst is a bit non-sensical IMO.
And by this point since it is starting fresh it would be best to use vst2 instead of simply vst anyway.

**Comment by hfiguiere** [comment]
No we can't change the path. But VST_PATH is set, what the problem? It works fine in LMMS whose flatpak has been shipping Carla since day one to support plugins.

**Comment by falkTX** [comment]
It is just wrong. Does flatpak work on BSD? maybe not now, but eventually could, are they going to call it bsdvst then?
Doesn't make any sense to make a distinction for the vst2 format like this.

**Comment by hfiguiere** [comment]
does it break Carla?

**Comment by falkTX** [comment]
If we use `~/.lxvst` (see #1, the local vs system paths usage is inconsistent) then I would say yes it does.

```
$ find ~/.vst -name "*.so" | wc -l
7

$ find ~/.lxvst -name "*.so" | wc -l
find: '/home/falktx/.lxvst': No such file or directory
0
```

**Comment by hfiguiere** [comment]
it doesn't set `~/.lxvst` anywhere.

**Comment by hfiguiere** [comment]
the system path is IMMUTABLE

**Comment by falkTX** [comment]
> it doesn't set ~/.lxvst anywhere.

Yes but it should, that is why I link to #1
Either it allows plugins from the local folder or not, right now it is inconsistent because it is allowed for LV2 but not others.

The system path should really be changed IMO.
On the next version of freedesktop-linuxaudio-plugins of course, not on the existing one.
The longer it goes without the change the harder it will be later on to reverse course.

**Comment by hfiguiere** [comment]
just to clarify. **There is no standard**. There is just a solution to make things work around something that was designed for it, and any other flatpak has to follow it.

It was originally done because I was flatpaking Ardour, LMMS came second in the pipe and I tested too, notably because it was the only to have VST3 support **via Carla**.

**Comment by falkTX** [comment]
I get it, but now that flatpak is going ahead with packaging, flatpak should be the one setting the standard for this.
We cannot even officially make new VST2 plugins anymore, we are on our own so we define it whatever we want/need.
So it really seems best to go in the correct path this time and not keep things as-is for the sake of legacy, we can make it follow better rules than before.

I would go even further to say that, with VST3 now being the focus of new plugins/developers, having lxvst or simply just vst is in itself wrong.
The format and paths should use VST2 name. In case you are not aware VST2 and VST3 are completely different and incompatible, calling something VST and something else VST3 is bound to generate confusion with users. (because at the end of the day a VST3 is a VST from the user's POV)

Since this is for flatpak packaging, strict rules can and should be defined.
You can define new rules and patch old software to follow these new rules.

For example Carla already accepts VST2_PATH as alternative to VST_PATH.

I really want to stress out that getting this properly defined is essential at this point before even more software is added and packaged in flathub.

**Comment by Bleuzen** [comment]
I had a look and so far saw only Ardour using `lxvst`. For other packages of vst plugins (aside from flathub) just `vst` seems to be more common. This name also makes more sense in my opinion. So if it is still possible, I would vote for changing the naming. Should be doable with next runtime/sdk branch, since we have to make changes anyway, right?

But on the other hand, even if this won't be done, I'm also fine with it.
Tbf, it isn't a dealbreaker as is.

If only the user directory is a problem, we could simply use:

`"--env=VST_PATH=$HOME/.vst:/app/extensions/Plugins/lxvst"`

So in the end, while it would be nice, if it is too much work, I wouldn't care too much.

---

Anyway, this has nothing to do with Carla, but with flathub plugin packaging. So we should probably open a new ticket on the flathub or plugin base extension issue tracker for further discussion.

**Comment by Bleuzen** [comment]
> For example Carla already accepts VST2_PATH as alternative to VST_PATH.

Oh and I really like this one. I also think the vst = vst2 thing is confusing.
But could lead to problems on older apps, so most likely won't happen.

**Comment by hfiguiere** [comment]
> If only the user directory is a problem, we could simply use:
> 
> `"--env=VST_PATH=$HOME/.vst:/app/extensions/Plugins/lxvst"`

you can't, `$HOME` isn't expanded and it's unlikely to be fixed.
(one need to us a wrapper script to set the environment)

**Comment by falkTX** [comment]
> > For example Carla already accepts VST2_PATH as alternative to VST_PATH.
> 
> Oh and I really like this one. I also think the vst = vst2 thing is confusing. But could lead to problems on older apps, so most likely won't happen.

But older apps are being repackaged into flathub, no? So it would make sense to patch them.
I think the small effort is worth it, taking into account long term benefits.


And since Carla still has the `lxvst` in its properties, which I considered to be wrong, I would kindly ask to keep this ticket open.

**Comment by Bleuzen** [comment]
Just created a new ticket for it:
https://github.com/flathub/org.freedesktop.LinuxAudio.BaseExtension/issues/10

> you can't, `$HOME` isn't expanded and it's unlikely to be fixed. (one need to use a wrapper script to set the environment)

Yeah I know now, this was only a example for the naming.

> But older apps are being repackaged into flathub, no? So it would make sense to patch them.
I think the small effort is worth it, taking into account long term benefits.

This is what we should discuss in the new ticket on the base extension.

> And since Carla still has the lxvst in its properties, which I consider to be wrong, I would kindly ask to keep this ticket open.

All apps using plugins currently have this. Wouldn't make sense to have a bunch of tickets for one issue, that isn't even the apps fault.

I added a checklist of apps to the newly created ticket to track them.

**Comment by falkTX** [comment]
reopening since the alternative/next issue was basically closed as "wont fix".
for me, this seems critical. we should start with a good foundation, not just go randomly packaging things.

**Comment by Bleuzen** [comment]
@falkTX makes no sense to reopen here.
There is no way I can do this only for Carla. The plugins path is fixed and set by https://github.com/flathub/org.freedesktop.LinuxAudio.BaseExtension.
If I change it for the Carla flatpak, Carla simply won't be able to see VST plugins anymore. So I can't change it here.
And I don't like having open issues that I can't do anything about.
If this will be changed, it **must** be changed in the BaseExtension first.

**Comment by falkTX** [comment]
Well if the BaseExtension does not change, this can't change either and thus can't be fixed.
Can understand not wanting to have such issues open, but having them closed is not ideal either.

My take is that attention needs to be brought up to the issue and not simply ignored. As long as the issue remains, it should be kept open, even if not fault from this side.
"Out of sight, out of mind" should not be an approach taken to software issues.

**Comment by Bleuzen** [comment]
solved since switch to 22.08 runtime
